PublicMapping/district-builder-2
# Add healthcheck endpoint to backend

## Overview

Adds a healthcheck endpoint that checks for database connectivity.

### Demo

```
$ curl http://localhost:3005/healthcheck
{"status":"ok","info":{"database":{"status":"up"}},"details":{"database":{"status":"up"}}}
```

### Notes

This is using the `terminus` healthcheck library, which Nest has an existing integration for (https://docs.nestjs.com/recipes/terminus). This library supports [a few builtin health checks](https://github.com/nestjs/terminus/tree/f62c2b2d9170811ad9221469fd286e0653d6667a/lib/health-indicators) and will also allow us to write our own if necessary.

For now I've only enabled the DB connection check, but I imagine the DNS, disk space or available memory checks could be useful as well.

## Testing Instructions

- `scripts/update`
- `scripts/server`
- http://localhost:3005/healthcheck should return a 200 status and the JSON returned should show `{"database":{"status":"up"}}`
- In a separate tab from where `scripts/server` is running, run `docker-compose stop database`
- http://localhost:3005/healthcheck should return a 503 status and the JSON returned should show `{"database":{"status":"down"}}`

Closes #6
